File: app/src/main/java/com/glucocontrol/domain/model/GlucoseReading.kt
package com.glucocontrol.domain.model

import java.time.LocalDate
import java.time.LocalTime

/**
 * Modelo de dominio que representa una medición de glucosa en sangre.
 *
 * Es la única fuente de verdad para la lógica de negocio.
 * La capa de datos transforma esta clase hacia/desde [GlucoseReadingEntity].
 * Este modelo no contiene ninguna dependencia de Android.
 *
 * @param id        Identificador único. Valor 0 indica que el registro aún no existe en BD.
 * @param date      Fecha de la medición.
 * @param time      Hora de la medición. Nullable para registros históricos donde no se conoce la hora.
 * @param valueMgDl Valor de glucosa en mg/dL.
 * @param tag       Etiqueta de contexto clínico (ayunas, post-comida, etc.).
 * @param notes     Texto libre para anotar anomalías (estrés, ejercicio, tipo de comida). Nullable.
 */
data class GlucoseReading(
    val id: Long = 0,
    val date: LocalDate,
    val time: LocalTime? = null,
    val valueMgDl: Int,
    val tag: ReadingTag = ReadingTag.OTRO,
    val notes: String? = null,
)

File: lib/domain/model/glucose_reading.dart
import 'package:glucocontrol/domain/model/reading_tag.dart';

class GlucoseReading {
  final int id;
  final DateTime date;
  final DateTime? time;
  final int valueMgDl;
  final ReadingTag tag;
  final String? notes;

  const GlucoseReading({
    this.id = 0,
    required this.date,
    this.time,
    required this.valueMgDl,
    this.tag = ReadingTag.otro,
    this.notes,
  });

  GlucoseReading copyWith({
    int? id,
    DateTime? date,
    DateTime? time,
    bool clearTime = false,
    int? valueMgDl,
    ReadingTag? tag,
    String? notes,
    bool clearNotes = false,
  }) {
    return GlucoseReading(
      id: id ?? this.id,
      date: date ?? this.date,
      time: clearTime ? null : (time ?? this.time),
      valueMgDl: valueMgDl ?? this.valueMgDl,
      tag: tag ?? this.tag,
      notes: clearNotes ? null : (notes ?? this.notes),
    );
  }
}
